sycamore/scans/file_scan.py

```diff
@@ -103,13 +103,15 @@ def __init__(
         parallelism: Optional[int] = None,
         filesystem: Optional[FileSystem] = None,
         metadata_provider: Optional[FileMetadataProvider] = None,
+        filter_paths_by_extension: bool = True,
```

Comment:
Nice, I wonder if there's some sort of warning/gotcha the docs need for this so one doesn't assume this means the pipeline will handle multiple file types. If the directory has htmls and pdfs the partitioner will likely break?

Reply:
If you set this and run multiple file types through, yes it will probably break. Added a note in the review-fix-1 commit. Let me know if the note is clear enough.

All the scripts we write seem to have pipelines for pdf and pipelines for html.

Reply:
It broke for me.

Reply:
(not a comment on this PR, but a response to Vinayak's comment.)